packages/core/nuxt-module/lib/plugins/i18n-cookies.js

```diff
@@ -70,19 +77,27 @@ const i18nCookiesPlugin = ({ $cookies, i18n, app, redirect }) => {
     throw new Error(`Following fields are missing in the i18n configuration: ${missingFields.join(', ')}`);
   }
 
-  if (cookieLocale !== settings.locale) {
+  if ((cookieLocale !== settings.locale && autoChangeCookie.locale) || !cookieLocale) {
     $cookies.set(cookieNames.locale, settings.locale, cookieOptions);
   }
 
-  if (cookieCurrency !== settings.currency) {
+  if ((cookieCurrency !== settings.currency && autoChangeCookie.currency) || !cookieCurrency) {
     $cookies.set(cookieNames.currency, settings.currency, cookieOptions);
   }
 
-  !$cookies.get(cookieNames.country) && $cookies.set(cookieNames.country, settings.country, cookieOptions);
+  if((cookieCountry !== settings.country && autoChangeCookie.country) || !cookieCountry){
+    $cookies.set(cookieNames.country, settings.country, cookieOptions);
+  }
 
   i18n.onBeforeLanguageSwitch = (oldLocale, newLocale, isInitialSetup, context) => {
-    $cookies.set(cookieNames.locale, newLocale, cookieOptions);
-    $cookies.set(cookieNames.currency, getCurrencyByLocale(newLocale), cookieOptions);
+    if(autoChangeCookie.locale){
```

Comment:
Without default values set to `true` this will not meet condition and will change default behavior. It may cause BC.